lib/optimizely/event/batch_event_processor.rb

```diff
@@ -90,56 +87,38 @@ def process(user_event)
         return
       end
 
-      @mutex.synchronize do
-        begin
-          @event_queue << user_event
-          @received.signal
-        rescue Exception
-          @logger.log(Logger::WARN, 'Payload not accepted by the queue.')
-          return
-        end
+      begin
+        @event_queue.push(user_event, true)
```

Comment:
Setting 2nd arg to true so that the main thread doesn't block and an exception is raised if the Queue is full. 
https://ruby-doc.org/core-2.3.0_preview1/SizedQueue.html#method-i-push

Reply:
can you please make sure in other SDKs, we are also getting queue size, but here we are just setting default queue size. Can you please add attribute to support that.

Reply:
We are already allowing user to pass it's own queue. Like we do in spec file, send our own SizedQueue with different size.